src/syscheckd/registry/registry.c

```diff
@@ -0,0 +1,501 @@
+    fim_tmp_file *file;
+    registry *configuration;
+
+    configuration = fim_registry_configuration(data->registry_entry.key->path, data->registry_entry.key->arch);
+
+    if (alert && configuration) {
+        cJSON *json_event = fim_registry_event(data, NULL, configuration, event_mode, FIM_DELETE, NULL, NULL);
+
+        if (json_event) {
+            char *json_formated = cJSON_PrintUnformatted(json_event);
+            send_syscheck_msg(json_formated);
+            os_free(json_formated);
+
+            cJSON_Delete(json_event);
+        }
+    }
+
+    if (fim_db_get_values_from_registry_key(fim_sql, &file, FIM_DB_DISK, data->registry_entry.key->id) == FIMDB_OK) {
+        if (file && file->elements) {
+            fim_db_process_read_registry_data_file(fim_sql, file, mutex, fim_registry_process_value_delete_event,
+                                                   FIM_DB_DISK, _alert, _ev_mode, _w_evt);
+        }
+    }
+
+    fim_db_remove_registry_key(fim_sql, data);
+}
+
+void fim_registry_process_unscanned_entries() {
+    fim_tmp_file *file;
+    fim_event_mode event_mode = FIM_SCHEDULED;
+
+    if (fim_db_get_registry_keys_not_scanned(syscheck.database, &file, FIM_DB_DISK) == FIMDB_OK) {
+        if (file && file->elements) {
+            fim_db_process_read_file(syscheck.database, file, FIM_TYPE_REGISTRY, &syscheck.fim_registry_mutex,
+                                     fim_registry_process_key_delete_event, FIM_DB_DISK, &_base_line, &event_mode, NULL);
+        }
+    } else {
+        mwarn("Failed to get unscanned registry keys");
+    }
+
+    if (fim_db_get_registry_data_not_scanned(syscheck.database, &file, FIM_DB_DISK) == FIMDB_OK) {
+        if (file && file->elements) {
+            fim_db_process_read_registry_data_file(syscheck.database, file, &syscheck.fim_registry_mutex,
+                                                   fim_registry_process_value_delete_event, FIM_DB_DISK, &_base_line,
+                                                   &event_mode, NULL);
+        }
+    } else {
+        mwarn("Failed to get unscanned registry values");
+    }
+}
+
+void fim_registry_process_value_event(fim_entry *new, fim_entry *saved, int arch, fim_event_mode mode, BYTE *data_buffer) {
+    char value_path[MAX_KEY + 2];
+    registry *configuration;
+    cJSON *json_event;
+    char *diff = NULL;
+
+    snprintf(value_path, MAX_KEY, "%s\\%s", new->registry_entry.key->path, new->registry_entry.value->name);
+
+    configuration = fim_registry_configuration(value_path, arch);
+    if (configuration == NULL) {
+        return;
+    }
+
+    if (fim_registry_validate_path(value_path, configuration)) {
+        return;
+    }
+
+    if (fim_check_restrict(value_path, configuration->filerestrict)) {
+        return;
+    }
+
+    saved->registry_entry.value =
+    fim_db_get_registry_data(syscheck.database, new->registry_entry.key->id, new->registry_entry.value->name);
```

Comment:
Please, move the function to the line above.
```suggestion
    saved->registry_entry.value = fim_db_get_registry_data(syscheck.database, new->registry_entry.key->id,
                                                           new->registry_entry.value->name);
```

Reply:
Done